### PR TITLE
Add Ui Task List panel.

### PR DESCRIPTION
**Pull request by Nephelite**
Warning: This current iteration will cause many merge conflicts with imminent changes from Logic and Model. DO NOT MERGE!

### CODE REVIEW

**Comment by Nephelite** [comment]
All necessary changes have been implemented. Ready for reviewing now.

**Comment by seetohjinwei** [comment]
Should we implement `logic.getTaskList()` and `task.getStatus()` first before merging this in?

Also, I think the temporary task list probably shouldn't be merged in too.

**Comment by dexter-sim** [comment]
Do we want TaskList to be permanently displayed beside contact list?

**Comment by Nephelite** [comment]
> Should we implement `logic.getTaskList()` and `task.getStatus()` first before merging this in?
> 
> Also, I think the temporary task list probably shouldn't be merged in too.

The temporary task list is needed until logic has a getTaskList method for me to obtain the task list. This is something the logic side can change (or me) once they are done with their part. I already have the method needed, so the only change needed in Ui is just removing the /* */ and the temporary task list.

File: src/main/java/seedu/address/ui/TaskListCard.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    private Label assignment;
+    @FXML
+    private Label description;
+    @FXML
+    private Label isDone;
+    @FXML
+    private Label id;
+
+    /**
+     * Creates a {@code PersonCode} with the given {@code Person} and index to display.
```
Comment (dexter-sim) [inline src/main/java/seedu/address/ui/TaskListCard.java:40]: I think you may have forgotten to update the javadoc for this method

File: src/main/java/seedu/address/ui/TaskListPanel.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    private static final String FXML = "TaskListPanel.fxml";
+    private final Logger logger = LogsCenter.getLogger(TaskListPanel.class);
+
+    @FXML
+    private ListView<Task> taskListView;
+
+    /**
+     * Creates a {@code PersonListPanel} with the given {@code ObservableList}.
+     */
+    public TaskListPanel(ObservableList<Task> taskList) {
```
Comment (dexter-sim) [inline src/main/java/seedu/address/ui/TaskListPanel.java:26]: Same as the constructor in `TaskListCard.java`

**Review by dexter-sim — CHANGES_REQUESTED** [review]
You might have forgotten to update the javadoc for some methods, but otherwise, the code looks good.

File: src/main/java/seedu/address/ui/TaskListCard.java (hunk 1 above)
Comment (Nephelite) [reply src/main/java/seedu/address/ui/TaskListCard.java]: Donezo

File: src/main/java/seedu/address/ui/TaskListPanel.java (hunk 2 above)
Comment (Nephelite) [reply src/main/java/seedu/address/ui/TaskListPanel.java:26]: Donezo

File: src/main/java/seedu/address/ui/TaskListCard.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import seedu.address.model.task.Task;
+
+/**
+ * An UI component that displays information of a {@code Person}.
+ */
+public class TaskListCard extends UiPart<Region> {
```
Comment (dexter-sim) [inline src/main/java/seedu/address/ui/TaskListCard.java:12]: I think there is another javadoc to be updated here.

**Review by dexter-sim — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/ui/TaskListCard.java (hunk 3 above)
Comment (Nephelite) [reply src/main/java/seedu/address/ui/TaskListCard.java]: done

**Pull request merged**